### PR TITLE
Add subs upgrade downgrade options

### PR DESCRIPTION
**Pull request by joycetoh8**
Note that default proration mode is set to `DEFERRED` and for a subscription upgrade it is set to `IMMEDIATE_AND_CHARGE_PRORATED_PRICE`.

### CODE REVIEW

File: src/js/lib/play-billing.js [hunk 1]
```diff
@@ -166,6 +167,9 @@ export class PlayBillingService {
         supportedMethods: this.serviceURL,
         data: {
           sku: sku.itemId,
+          oldSku: oldPurchase?.itemId,
+          purchaseToken: oldPurchase?.purchaseToken,
+          prorationmode: 'immediateAndChargeProratedPrice',
```
Comment (nohe427) [inline src/js/lib/play-billing.js:172]: Can we add a document comment on what the other options are for prorationmode? (Or at least a link to the documentation?)

File: src/js/components/sku-list.js [hunk 2]
```diff
@@ -48,21 +49,56 @@ class SkuList extends LitElement {
...
+    } else {
+      return this.type;
+    }
+  }
+
   /**
    * @return {TemplateResult}
    */
   render() {
+    const basicSubPurchase = this.purchases.find((purchase) => purchase.itemId === BASIC_SUB);
```
Comment (nohe427) [inline src/js/components/sku-list.js:74]: Are these just boolean values? Can we maybe set theses to hasBasicSubPurchase?

File: src/js/components/sku-list.js [hunk 3]
```diff
@@ -48,21 +49,56 @@ class SkuList extends LitElement {
...
         price="${this.service.getSkuPrice(sku, this.locale)}"
         .purchase=${purchase
           ? null
           : async function () {
-              const { response, valid } = await this.service.purchase(sku.itemId);
+              let purchaseMade;
+              switch (skuType) {
+                case 'downgrade':
+                  // downgrade from premium to basic subscription
+                  purchaseMade = await this.service.purchase(sku.itemId, premiumSubPurchase);
```
Comment (nohe427) [inline src/js/components/sku-list.js:91]: is premiumSubPurchase here the premium subscription token?

File: src/js/lib/play-billing.js (hunk 1 above)
Comment (joycetoh8) [reply src/js/lib/play-billing.js:172]: Yes, good call. I will add a comment.

File: src/js/components/sku-list.js (hunk 2 above)
Comment (joycetoh8) [reply src/js/components/sku-list.js:74]: No, these will be the actual `PurchaseDetails` for the basic and premium subs.

File: src/js/components/sku-list.js (hunk 3 above)
Comment (joycetoh8) [reply src/js/components/sku-list.js:91]: It's the whole `PurchaseDetails`. In play-billing.js lines 170-171, we need to pass in the purchase item ID and purchase token there. So, I pass in the whole `PurchaseDetails` to the `purchase` method.

File: src/js/lib/play-billing.js (hunk 1 above)
Comment (joycetoh8) [reply src/js/lib/play-billing.js]: DONE

**Review by nohe427 — APPROVED** [review]

**Pull request merged**